src/Platform/Microsoft.Testing.Platform/ServerMode/JsonRpc/PerRequestServerDataConsumerService.cs

```diff
@@ -216,22 +216,24 @@ public async Task ConsumeAsync(IDataProducer dataProducer, IData value, Cancella
             return;
         }
 
-        if (value is TestNodeUpdateMessage update)
+        switch (value)
         {
-            await ProcessTestNodeUpdateAsync(update, cancellationToken);
-            PopulateTestNodeStatistics(update);
-        }
-        else if (value is SessionFileArtifact sessionFileArtifact)
-        {
-            Artifacts.Add(new Artifact(sessionFileArtifact.FileInfo.FullName, dataProducer.Uid, FileType, sessionFileArtifact.DisplayName, sessionFileArtifact.Description));
-        }
-        else if (value is FileArtifact file)
-        {
-            Artifacts.Add(new Artifact(file.FileInfo.FullName, dataProducer.Uid, FileType, file.DisplayName, file.Description));
-        }
-        else if (value is TestNodeFileArtifact testNodeFileArtifact)
-        {
-            Artifacts.Add(new Artifact(testNodeFileArtifact.FileInfo.FullName, dataProducer.Uid, FileType, testNodeFileArtifact.DisplayName, testNodeFileArtifact.Description));
+            case TestNodeUpdateMessage update:
+                await ProcessTestNodeUpdateAsync(update, cancellationToken);
+                PopulateTestNodeStatistics(update);
+                break;
+
+            case TestNodeFileArtifact testNodeFileArtifact:
```

Comment:
This inherits from `SessionFileArtifact` so we need to handle subclass before base class.